javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function called addTwo that takes in an array and adds two to every value using a for loop. Place the new value in a new array. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const addTwo = (arr) => {
+  const output = [];
+  for(let num of arr) {
+    output.push(num + 2);
+  }
+  return output;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named typeNum that, given an array as input, uses filter to return an array containing only the numbers.
+
+For example, typeNum([1, 'bob' ,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const typeNum = (arr) => {
+  return arr.filter(any => typeof any === 'number');
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named containsAnd that, given an array of strings as input, uses filter to return an array containing only strings that contain 'and' within the string.
+
+For example, containsAnd(['panda', 'ran', 'and']) returns ['panda', 'and'].
+------------------------------------------------------------------------------------------------ */
+
+const containsAnd = (arr) => {
+  return arr.filter(str => str.includes('and'));
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named oddValues that, given an array of integers as input, uses filter to return an array containing only the odd integers.
+
+For example, oddValues([1,2,3]) returns [1,3].
+------------------------------------------------------------------------------------------------ */
+
+const oddValues = (arr) => {
+  return arr.filter(num => num % 2);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named notInFirstArray that, given two arrays as input, uses filter to return an array of all the elements in the second array that are not included in the first array.
+
+For example, notInFirstArray([1,2,3], [1,2,3,4]) returns [4].
+------------------------------------------------------------------------------------------------ */
+
+const notInFirstArray = (forbiddenValues, arr) => {
+  return arr.filter(value => !(forbiddenValues.includes(value)));
+};
```

Comment:
You actually don't need those parenthesis around your "forbiddenValue.includes" bit. Doesn't hurt, but also not necessary.